elide-core/src/main/java/com/yahoo/elide/graphql/DeferredId.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2017, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.graphql;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.yahoo.elide.core.PersistentResource;
+import lombok.Getter;
+
+import java.io.IOException;
+
+/**
+ * Simple encapsulation to serialize the id field from a {@link PersistentResource} object.
```

Comment:
Add the following text.

The id for any given entity might be populated at transaction commit (as opposed to inline with the data fetch).  This class wraps a PersistentResource and allows deferred deserialization of the ID field until when it is populated and when the GraphQL response is generated.